In [1]:
import numpy as np
import scipy.stats as st

Средние расходы на обследования одного больного до и после ознакомления с мнением коллег.
Определить количество выборок - у нас две
Определить зависимость выборок - у нас зависимые

Используем критерий Уилкоксона
alpha = 0.05 > pvalue=0.004741768038406972
Различия есть. Мнение повлияло

In [6]:
X = np.array([20,17, 14, 42, 50, 62, 8, 49, 81, 54, 48, 55, 56])
Y = np.array ([20, 26, 1, 24, 1, 47, 15, 7, 65, 9, 21, 36, 30])

st.wilcoxon(X, Y)

WilcoxonResult(statistic=3.0, pvalue=0.004741768038406972)

WilcoxonResult(statistic=3.0, pvalue=0.004741768038406972)

alpha = 0.05 > pvalue=0.004741768038406972
Различия есть. Мнение повлияло

Расходы на лечения

In [8]:
X1 = np.array([ 32, 41, 51, 29, 76, 47, 60, 58, 40, 64, 73, 66, 73])
Y1 = np.array ([42, 90, 71, 47, 56, 43, 137, 63, 28, 60, 87, 69, 50])

st.wilcoxon(X1, Y1)

WilcoxonResult(statistic=31.5, pvalue=0.339599609375)

WilcoxonResult(statistic=31.5, pvalue=0.339599609375)
alpha = 0.05 < pv
Следовательно статистически значимых различий нет


In [9]:
gr1 =  np.array([0.5, 0.7, 1, 1.2, 1.4])
gr2 =  np.array([1.3, 1.45, 1.6, 1.7, 1.8])
gr3 =  np.array([6.2, 12.6, 13.2, 14.1, 14.2])

st.kruskal(gr1, gr2, gr3)

KruskalResult(statistic=12.02000000000001, pvalue=0.002454088180413905)

KruskalResult(statistic=12.02000000000001, pvalue=0.002454088180413905)
pv < alpha -> статистически значимые различия есть